### ISSUE TITLE
instapush: non ascii characters cause an exception Parameter status cannot be formatted

### ISSUE THREAD

**Issue by AlexBGoode** [jpmens/mqttwarn#154]
Hello

I managed to setup subscribe to mosquitto and instapush works as needed until I start using international messages, i.e. messages with non ascii characters

In my case it is Russian and when I send a message:
   **mosquitto_pub -h 127.0.0.1 -t test -m '{"status":"Привет"}'**
then I get the following error:
   **DEBUG [instapush] Parameter status cannot be formatted: 'ascii' codec can't encode characters in position 0-5: ordinal not in range(128)**

Tried with German just for example:
   **mosquitto_pub -h 127.0.0.1 -t test -m '{"status":"Straße"}'**
then I get the following error
   **DEBUG [instapush] Parameter status cannot be formatted: 'ascii' codec can't encode character u'\xdf' in position 4: ordinal not in range(128)**

Any ideas what's wrong with that? I believe most of us have to use non typical ascii characters

**Comment by jpmens** [comment]
I'm ashamed to say, I don't think I've tested most of the modules with UTF-8. I _assume_ (and hope) the problem is within the instapush module.  I cannot verify this now, but maybe you can have a look and/or (preferably) check with the "log" module whether your UTF-8 is correctly passed into the target file.

**Comment by AlexBGoode** [comment]
Sure. I did and found out that all my lovely UTF-8 messages are successfully appeared in mosquitto's log, in the log of mqttwarn in general (during pre-processing) and also in log-destination section:

*bash-3.2$ cat mqttwarn.log*
2015-11-13 16:35:11,525 DEBUG [mqttwarn] Message received on failover: **Привет 1**
2015-11-13 16:35:11,526 DEBUG [mqttwarn] Section [failover] matches message on failover. Processing...
2015-11-13 16:35:11,526 DEBUG [mqttwarn] Message on failover going to log:error
2015-11-13 16:35:11,526 DEBUG [mqttwarn] New `log:error' job: failover
2015-11-13 16:35:11,526 DEBUG [mqttwarn] Message on failover going to file:mqttwarn
2015-11-13 16:35:11,527 DEBUG [mqttwarn] Processor is handling: `log' for error
2015-11-13 16:35:11,527 DEBUG [mqttwarn] New `file:mqttwarn' job: failover
2015-11-13 16:35:11,528 DEBUG [log] * * * MODULE=services/log.pyc: service=log, target=error
2015-11-13 16:35:11,528 ERROR [log] **Привет 1**
2015-11-13 16:35:11,528 DEBUG [mqttwarn] Processor is handling: `file' for mqttwarn
2015-11-13 16:35:11,529 DEBUG [file] * * * MODULE=services/file.pyc: service=file, target=mqttwarn

Also the file-destination works well and UTF-8 messages are recorder with no problem:

*bash-3.2$ cat mqttwarn.err*
**Привет 1**

**Comment by jpmens** [comment]
So, the problem seems to be in on line 30 of `services/instapush.py`.

Could you change that line in your mqttwarn installation to remove the `.encode('utf-8')` at the end so that the line reads:

```python
data["trackers"][key] = data["trackers"][key].format(**item.data)
```

I'm just stabbing in the dark to be honest. ;-)

**Comment by amotl** [comment]
Dear @AlexBGoode,

your issue might have been resolved through `mqttwarn 0.12.0`. You should be able to install this using `pip install mqttwarn` on Python2.

With kind regards,
Andreas.

**Comment by amotl** [comment]
It looks like Instapush went out of service. So, I am closing this.
